tutorial/markdown/java/spring-data/spring-data.md

```diff
@@ -0,0 +1,427 @@
+
+These detailed workflows provide a comprehensive understanding of how each mapping is handled by the controller, service, and repository components in your Spring Data project.
+
+## Custom SQL++ Queries
+
+The `AirlineRepository` interface contains a `@Query` annotation that allows us to create custom N1QL queries. The `@ScanConsistency` annotation allows us to specify the scan consistency for the query. The `@Param` annotation allows us to specify parameters for the query.
+
+```java
+@Query("#{#n1ql.selectEntity} WHERE #{#n1ql.filter} AND country = $country")
+@ScanConsistency(query = QueryScanConsistency.REQUEST_PLUS)
+List<Airline> findByCountry(@Param("country") String country);
+```
+
+> _from repository/AirlineRepository.java_
+
+The `findByCountry` method returns a list of airlines by country. It uses the `@Query` annotation to create a custom N1QL query. The `@Param` annotation is used to specify the `country` parameter for the query. The `@ScanConsistency` annotation is used to specify the scan consistency for the query.
+
+```java
+@Query("#{#n1ql.selectEntity} WHERE #{#n1ql.filter} AND ANY destination IN routes SATISFIES destination = $airport END")
+@ScanConsistency(query = QueryScanConsistency.REQUEST_PLUS)
+List<Airline> findByDestinationAirport(@Param("airport") String airport);
+```
+
+> _from repository/AirlineRepository.java_
+
+The `findByDestinationAirport` method returns a list of airlines by destination airport. It uses the `@Query` annotation to create a custom N1QL query. The `@Param` annotation is used to specify the `airport` parameter for the query. The `@ScanConsistency` annotation is used to specify the scan consistency for the query.
+
+<!-- Link to docs -->
+
+For more information, see the [Couchbase Java SDK documentation](https://docs.couchbase.com/java-sdk/current/howtos/n1ql-queries-with-sdk.html).
+
+## Running The Tests
+
+To run the tests, execute `./gradlew test` (`./gradlew.bat test` on Windows).
+
+```sh
+./gradlew test
+```
+
+## Project Setup Notes
+
+This project was created using the [Spring Initializr](https://start.spring.io/) with the following options:
+
+- Project: Gradle Project
+- Language: Java
+- Spring Boot: 2.7.18
+- Packaging: Jar
+- Java: 8
+- Dependencies: Spring Web, Spring Data Couchbase, Springdoc OpenAPI UI
+
+## Contributing
+
+Contributions are welcome! If you'd like to contribute to this project, please fork the repository and create a pull request.
+
+## Appendix
+
+### Extending API by Adding New Entity
+
+If you would like to add another entity to the APIs, these are the steps to follow:
+
+- Create the new entity (collection) in the Couchbase bucket. You can create the collection using the [SDK](https://docs.couchbase.com/sdk-api/couchbase-java-client-3.5.2/com/couchbase/client/java/Collection.html#createScope-java.lang.String-) or via the [Couchbase Server interface](https://docs.couchbase.com/cloud/n1ql/n1ql-language-reference/createcollection.html).
```

Comment:
The SDK link should point to the right Management API

Reply:
updated the SDK link